src/release/breaking-changes/boxdecoration-shape-migration.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Migration guide for BoxDecoration.circle
+description: Removal of BoxDecoration.circle and how to migrate
+---
+
+## Summary
+
+This change removes `BoxDecoration(shape: BoxShape.circle)`. Existing apps
+should use `ShapeDecoration(shape: CircleBorder())` instead.
+
+## Context
+
+`BoxDecoration(shape: BoxShape.circle)` uses an enum. Each new enum value requires new code
+in many places. That new code may not shake out,
+since the compiler doesn't know which values never get used.
+`BoxDecoration` has a permanent binary size cost,
+even for people who don't use it.
+Flutter moved to `ShapeBorders` rather than `BoxShape` for new
+shapes, and the existence of `BoxShape` remains as a legacy feature.
+
+Use `ShapeDecoration`. It aligns with the rest of the Flutter API.
+`BoxShape` has been deprecated.
+Continuing to use that feature can cause confusion.
+
+## Description of change
+
+Remove `BoxDecoration(shape: BoxShape.circle)` and `BoxDecoration.shape`.
+
+## Migration guide
+
+- Replace `BoxDecoration(shape: BoxShape.circle)` with
+`ShapeDecoration(shape: CircleBorder())`.
+- Remove the `BoxDecoration.shape` property from
+`BoxDecoration(shape: BoxShape.rectangle)`.
+
+When you use `Border`,
+`CircleBorder(width: ..., color: ..., strokeAlign: ...)` inherits
+the `Border.all(width: ..., color: ..., strokeAlign: ...)` values.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+BoxDecoration(
+  shape: BoxShape.circle,
+)
+
+BoxDecoration(
+  shape: BoxShape.circle,
+  border: Border.all(width: 20, color: Colors.red),
+  boxShadow: [],
+)
+
+BoxDecoration(
+  shape: BoxShape.rectangle,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+ShapeDecoration(
+  border: CircleBorder(),
+)
+
+ShapeDecoration(
+  border: CircleBorder(width: 20, color: Colors.red),
+  shadows: [],
+)
+
+BoxDecoration(
+  shape: BoxShape.rectangle,
+)
+```
+
+## Timeline
+
+Landed in version: TBD<br>
+In stable release: not yet
+
+## References
+
+{% include docs/master-api.md %}
+
+API documentation:
+
+* [`BoxDecoration` (3.3.6)][]
+* [`BoxDecoration` (3.4.0)][]
+
+Relevant issues:
+
+* [Deprecate `BoxDecoration.shape` issue][]
+
+Relevant PRs:
+
+* [Deprecate `BoxDecoration.shape` PR][]
+
+<!-- Stable channel link: -->
+[`BoxDecoration` (3.3.6)]: {{site.api}}/flutter/lib/src/painting/BoxDecoration-class.html
+
+<!-- Master channel link: -->
+{% include docs/master-api.md %}
+
+[`BoxDecoration` (3.4.0)]: {{site.master-api}}/flutter/lib/src/painting/BoxDecoration-class.html
```

Comment:
These version numbers will go out of date every time we tag a build or cut a release. I think they can be omitted.